src/background/service/APIService.js

```diff
@@ -926,9 +926,9 @@ class APIService {
             if(myNotificationID === clickId){
                 let url
                 if(runtimeId){
-                    url = getExplorerUrl() + "paratimes/transactions/" + clickId
+                    url = getExplorerUrl() + "paratimes/transactions/" + encodeURIComponent(clickId)
```

Comment:
what's the format of clickId?

Reply:
oh no. It can be `hash?runtime=runtimeId`

Reply:
Refactored https://github.com/oasisprotocol/oasis-wallet-ext/blob/d6e14dec8d3e1ad1a5979e7bbc39b2775e7d1340/src/background/service/APIService.js#L921-L938

Reply:
what am I looking at here

Reply:
`url = getExplorerUrl() + "paratimes/transactions/" + encodeURIComponent(clickId)` was wrong because `encodeURIComponent('hash?runtime=runtimeId') === 'hash%3Fruntime%3DruntimeId'` (notifyId -> myNotificationID -> clickId)
```js
let notifyId = runtimeId ? hash +"?runtime="+runtimeId : hash
let myNotificationID
extension.notifications.create(notifyId, {...},(notificationItem)=>{ myNotificationID = notificationItem })
extension.notifications.onClicked.addListener(function (clickId) { if(myNotificationID === clickId){
  url = getExplorerUrl() + "paratimes/transactions/" + encodeURIComponent(clickId)
```

Force-pushed fix: https://github.com/oasisprotocol/oasis-wallet-ext/compare/6975b3c83c1110225738478e0a83428191576ccc..b07d9481e5e88f2834d53770913aff678ed92cbc

Additionally refactored:
- `myNotificationID` wasn't needed
- why create part of the url in two places:
  ```js
  let notifyId = runtimeId ?  encodeURIComponent(hash) +"?runtime="+encodeURIComponent(runtimeId) : encodeURIComponent(hash)
  notifyId -> myNotificationID -> clickId
    if(runtimeId){
        url = getExplorerUrl() + "paratimes/transactions/" + clickId
    }else{
        url = getExplorerUrl() + "transactions/" + clickId
    }

  ```
  Combined into
  ```js
  const notificationLinkAsId = runtimeId
      ? getExplorerUrl() + "paratimes/transactions/" + encodeURIComponent(hash) + "?runtime=" + encodeURIComponent(runtimeId)
      : getExplorerUrl() + "transactions/" + encodeURIComponent(hash)
  notificationLinkAsId -> notificationId
  ```
- `removeListener` was missing

Reply:
o nice